### ISSUE TITLE
criterion.weight in CheckPoint resnet50

### ISSUE THREAD

**Issue by kodinkod** [OML-Team/open-metric-learning#422]
When i want load my CheckPoint from train resnet50 with Arface i get    assert all(k.startswith(prefix) for k in state_dict.keys());
Becouse 'criterion.weight' in state_dict.  Can we find a bug fix and execute an extraction request: for example:
 if 'criterion.weight" in state_dict:
     del state_dict['criterion.weight']

in 128 row resnet.py

**Comment by kodinkod** [comment]
I have not done pull requests before , I will be glad to cooperate and help

**Comment by AlekseySh** [comment]
Hey, @kodinkod! First of all, thank you for using OML.

Before discussing a solution, could you please explain how did you create & train your model (please, provide code snippet or config.yaml). And also how did you load a state of your model? After that it will be more clear what should we do.

**Comment by kodinkod** [comment]
for train:
`dataframe_name: /
dataset_root: /
logs_root:  log/
logs_folder: ${now:%Y-%m-%d_%H-%M-%S}_${postfix} 
show_dataset_warnings: True

num_workers: 12
cache_size: 0

transforms_train:
  name: augs_hypvit_torch
  args:
    im_size: 224

transforms_val:
  name: norm_resize_hypvit_torch
  args:
    im_size: 224
    crop_size: 224


metric_args:
  metrics_to_exclude_from_visualization: [cmc,]
  cmc_top_k: [1, 3, 5]
  map_top_k: [1, 3, 5]
  pfc_variance: [0.5, 0.9, 0.99]
  return_only_main_category: True
  visualize_only_main_category: True

log_images: True
metric_for_checkpointing: OVERALL/cmc/1
extractor:
  name: resnet
  args:
    arch: resnet50
    weights: resnet50_imagenet1k_v1
    normalise_features: True
    remove_fc: False
    gem_p: 1



criterion:
  name: arcface
  args:
    smoothing_epsilon: 0.0
    m: 0.4
    s: 31
    in_features: 1000 # for resnet
    num_classes: 821

sampler:  # batch sampler, in this case
  name: balance
  args:
    n_labels: 13
    n_instances: 4

bs_train: 60
bs_val: 256
max_epochs: 10000
valid_period: 5

optimizer:
  name: adam
  args:
    lr: 1e-5

# There are several options for logging:
logger:
  name: tensorboard
  args:
    save_dir: lightning_logs/arcface/
    log_model_checkpoints: True`

in python:
` extractor_training_pipeline(cfg)`

for valid:
`bs_val: 256
num_workers: 8

transforms_val:
  name: norm_resize_albu
  args:
    im_size: 224

extractor:
  name: resnet
  args:
    arch: resnet50
    normalise_features: True
    remove_fc: False
    gem_p: 1
    weights: ../best.ckpt

metric_args:
  metrics_to_exclude_from_visualization: [cmc,]
  cmc_top_k: [1, 3, 5]
  map_top_k: [1, 3, 5]
  precision_top_k: [1,2,5]
  fmr_vals: [0.01]
  pfc_variance: [0.5, 0.9, 0.99]
  return_only_main_category: False
  visualize_only_main_category: True`

in python:
` extractor_validation_pipeline(cfg)`



The same problem when i Just write:
`extractor =  ResnetExtractor(weights='../best.ckpt',
                             arch="resnet50", normalise_features=True, remove_fc = False, gem_p=1).eval()`

**Comment by AlekseySh** [comment]
Thank you for the provided details! Seems like criterion has been saved by Lightning automatically (CheckpointCallback). There are two options.

1. Change saving behavior, probably we can configure CheckpointCallback, so it doesn't save criterion. It's a preferable option. 
2. Change loading behavior. For example, we can remove all the keys containing 'criterion' when loading model. The problem is that we need to do this in all the models. And it will not work for a custom model added by user to our registry. 


What are your thoughts on that?

PS. You are welcome to contribute. Please, check the contribution guide in Readme.

**Comment by kodinkod** [comment]
I think if we are sure that 'criterion' is not needed anywhere, then this gives us the right not to save it. I'll try to find the settings that allow you to do this.

**Comment by AlekseySh** [comment]
OML's extractors don't need `criterion` in a state dict for sure

**Comment by AlekseySh** [comment]
@kodinkod Hey! If you have time, we are still happy to see a contribution from you

**Comment by kodinkod** [comment]
I'm sorry, I was busy with work projects. I plan to use  def on_save_checkpoint(self, trainer: pl.Trainer, pl_module: pl.LightningModule, checkpoint). now I'm planning to do tests

**Comment by kodinkod** [comment]
@AlekseySh i make Pull requests
